src/onDeleteAccount.js

```diff
@@ -1,20 +1,47 @@
 const { log } = require('cozy-konnector-libs')
+const pLimit = require('p-limit')
 
 const getAccountId = require('./helpers/getAccountId')
 const CozyUtils = require('./CozyUtils')
 
-async function onDeleteAccount() {
+async function onDeleteAccount(accountId, cozyUtils) {
   log('info', 'onDeleteAccount: remove account id from the contact account')
-  const accountId = getAccountId()
-  const cozyUtils = new CozyUtils(accountId)
+
   const contactAccount = await cozyUtils.findContactAccount(accountId)
   if (contactAccount) {
     contactAccount.sourceAccount = null
-    await cozyUtils.client.save(contactAccount)
+    await cozyUtils.save(contactAccount)
+
+    // also remove all trashed contacts linked to this account
+    log(
+      'info',
+      'onDeleteAccount: remove all trashed contacts linked to this account'
+    )
+    const trashedContacts = await cozyUtils.getTrashedContacts(
+      contactAccount.id
+    )
+
+    const limit = pLimit(50)
+    const destroyPromises = trashedContacts.map(contact => () =>
+      cozyUtils.client.destroy(contact)
+    )
+
+    log('info', `'Will delete ${trashedContacts.length} contacts.`)
+    try {
+      await Promise.all(destroyPromises.map(limit))
+    } catch (err) {
+      log(
+        'error',
+        `Error encountered while deleting trashed contacts: ${err.message}`
+      )
+      throw err
+    }
   }
 }
 
-onDeleteAccount().then(
+const accountId = getAccountId()
+const cozyUtils = new CozyUtils(accountId)
```

Comment:
Any reason for moving these out of the function?

Reply:
for tests...

Reply:
Ah ok :p Well if we can revert that, the rest can be merged!

Reply:
I mean unit tests :-) I need to be able to pass the id and cozyUtils to the function to be able to write tests with ease

Reply:
Ah yes ok, my bad 👍